apps/prairielearn/src/lib/db-types.ts

```diff
@@ -15,6 +15,8 @@ export const CourseSchema = z.object({
   options: z.any(),
   path: z.string().nullable(),
   repository: z.string().nullable(),
+  sharing_name: z.string().nullable(),
+  sharing_token: z.string().nullable(),
```

Comment:
Would it be worth it to make this column non-nullable and add `DEFAULT gen_random_uuid()`?

Reply:
I can't really properly make `sharing_token` a non-null column unless we want to do a batched migration to backfill of all existing courses. (did you want to do that?).

My approach was going to be to just create and save it whenever someone visits the Sharing page for the first time.

It seems like Zod has some sort of feature where you can have a default value show up even if the value is not in the database. I'm not sure the value of using that in this case because if the `sharing_token` is actually null in the database, we need to know that piece of information so that we can save the one we create to the database. It doesn't seem like Zod supports this but I could be missing something.

Reply:
To clarify, when I say "make the column non-nullable and add a default", I'm referring to the database column itself, not the zod types. I should have put this comment on the database migration, not this file; this thought occurred to me when looking at this file, which is why the comment is here.

It isn't necessary to use a batched migration. The `pl_courses` table is small enough that we can just add a default at the schema level and let Postgres rewrite the <1000 course rows we have.

Reply:
sounds good, done.